<launch>

    <arg name="rviz_enable" default="false"/>
    <arg name="rosbag_enable" default="false"/>

    <arg name="xfer_format" default="2"/>
    <arg name="config" default="$(find livox_ros_driver)/config/livox_ros_driver.yaml"/>

    <node name="livox_lidar_publisher" pkg="livox_ros_driver" type="livox_ros_driver_node"
          required="true" output="screen">
        <rosparam command="load" file="$(arg config)"/>
        <param name="xfer_format" value="$(arg xfer_format)"/>
    </node>

    <group if="$(arg rviz_enable)">
        <node name="rviz" pkg="rviz" type="rviz" respawn="true"
                args="-d $(find livox_ros_driver)/config/display_lidar_points.rviz"/>
    </group>

    <group if="$(arg rosbag_enable)">
        <node pkg="rosbag" type="record" name="record" output="screen"
                  args="-a"/>
    </group>

</launch>
